packages/clerk-js/src/utils/captcha/turnstile.ts

```diff
@@ -112,59 +117,72 @@ async function loadCaptchaFromFAPIProxiedURL(fallbackUrl: string) {
  * - If the widgetType is 'smart', the captcha widget is rendered in a div with the id 'clerk-captcha'. If the div does
  *  not exist, the invisibleSiteKey is used as a fallback and the widget is rendered in a hidden div at the bottom of the body.
  */
-export const getTunstileToken = async (captchaOptions: {
-  siteKey: string;
-  scriptUrl: string;
-  widgetType: CaptchaWidgetType;
-  invisibleSiteKey: string;
-}) => {
-  const { siteKey, scriptUrl, widgetType, invisibleSiteKey } = captchaOptions;
-  let captchaToken = '',
-    id = '';
-  let isInvisibleWidget = !widgetType || widgetType === 'invisible';
+export const getTurnstileToken = async (opts: CaptchaOptions) => {
+  const { siteKey, scriptUrl, widgetType, invisibleSiteKey } = opts;
+  const { modalContainerQuerySelector, modalWrapperQuerySelector, closeModal, openModal } = opts;
+  const captcha: Turnstile = await loadCaptcha(scriptUrl);
+  const errorCodes: (string | number)[] = [];
+
+  let captchaToken = '';
+  let id = '';
   let turnstileSiteKey = siteKey;
+  let retries = 0;
+  let widgetContainerQuerySelector: string | undefined;
+  // The backend uses this to determine which turnstile to used in order to verify the token
+  let captchaWidgetTypeUsed: CaptchaWidgetType = null;
+
+  // modal
+  if (modalContainerQuerySelector && modalWrapperQuerySelector) {
+    captchaWidgetTypeUsed = 'smart';
```

Comment:
❗  How are we making sure that this is `smart` here? If the `siteKey` coming from dashboard is `invisible`, then it will use the invisible captcha here!!